README.md

```diff
@@ -115,11 +123,32 @@ The ‘role’ key maps to the ‘module:rRoleName’, which in this case is ‘
 
 The remaining key: value pairs map to the fields of the role as described in the STL file.
 
-### 3.4. test
+### 3.4. constants
+The ‘constants’ field must be a dictionary. Each entry in the dictionary defines a declared constant described in the stl_files listed in the ‘stl_files’ field. In the above example, there [example_base.stl](example/example_base.stl) contains the following:
+```
+example_base.stl:
+
+module example;
+...
+const string kHelloWorld;
+...
+```
+
+This value gets defined in the manifest here: 
+```
+'constants': {
+  'example::kHelloWorld': $hello_world,  # Must pass --manifest-args \
+                                         # 'hello_world="Hello, world!"'
+}
+```
+
+The key maps to the ‘<module>::<constantName>’, which in this case is ‘example::kHelloWorld’.
+
+### 3.5. test
 The ‘test’ field must be a list of strings naming the roles to be tested. Each entry in the test list must correspond to a ‘role’ value in the ‘roles’ list.
 
-### 3.5. Substituting Parameters
-Values can be substituted from the command line by passing a space separated list of key=value pairs to the --manifest-args option of test\_driver.py. The manifest .test file will then replace any occurence of $key with value. $key can be located anywhere in the manifest file (e.g. in the list of stl\_files, as a key or vparameter_example.test: test).
+### 3.6. Substituting Parameters
+Values can be substituted from the command line by passing a space separated list of key=value pairs to the --manifest-args option of test\_driver.py. The manifest .test file will then replace any occurence of $key with value. $key can be located anywhere in the manifest file (e.g. in the list of stl\_files, as a key or value in any of the manifest dictionaries).
```

Comment:
nit: Above you used the notation $<key> and <value>, but here it's still $key and value. Either way is fine, just make sure it's consistent